runtime/intern.c

```diff
@@ -699,8 +699,11 @@ static value intern_end(value res, mlsize_t whsize) {
 
   /* Memprof tracking has to be done here, because unmarshalling can
      still fail until now. */
-  if(block != NULL)
+  if(block != NULL) {
+    CAMLparam1(res);
     caml_memprof_track_interned(block, blockend);
+    CAMLdrop;
+  }
```

Comment:
This change should go away if `caml_memprof_track_interned` did not call callbacks directly as suggested bellow.